### PR TITLE
more code refactoring

### PR DESCRIPTION
**Pull request by Jacky-at-Slalom**
So what I’m trying to do is create a basic RESTful CRUD web service. The goal is to be able to hit the REST APIs exposed by the UserRepository with postman.

This cant be merge yet. I get a Process finished with exit code 1 error. I set sometime with Jesse tomorrow  to get his help on this.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,6 +36,39 @@
...
+			<artifactId>jaxb-api</artifactId>
+			<version>2.3.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.jetbrains.kotlin</groupId>
+			<artifactId>kotlin-stdlib-jdk8</artifactId>
+			<version>${kotlin.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.jetbrains.kotlin</groupId>
```
Comment (lynnSlalom) [inline pom.xml]: What does Kotlin do for you on this project? Please explain

File: src/main/java/com/javaproject/restservice/Application.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.javaproject.restservice;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
```
Comment (lynnSlalom) [inline src/main/java/com/javaproject/restservice/Application.java:6]: What does this annotation do?

File: src/main/java/com/javaproject/restservice/entity/User.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package com.javaproject.restservice.entity;
+
```
Comment (lynnSlalom) [inline src/main/java/com/javaproject/restservice/entity/User.java:2]: Only one line break needed

File: src/main/java/com/javaproject/restservice/entity/User.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+
+@Entity
+public class User {
+
+    @Id
```
Comment (lynnSlalom) [inline src/main/java/com/javaproject/restservice/entity/User.java:10]: What does @Id indicate?

File: src/main/java/com/javaproject/restservice/entity/User.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
```
Comment (lynnSlalom) [inline src/main/java/com/javaproject/restservice/entity/User.java:42]: What does the override do here?

File: src/main/java/com/javaproject/restservice/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package  com.javaproject.restservice.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import com.javaproject.restservice.entity.User;
+
+
+@RepositoryRestResource(path = "/users")
```
Comment (lynnSlalom) [inline src/main/java/com/javaproject/restservice/repository/UserRepository.java]: What does this do? @RepositoryRestResource?

Comment (lynnSlalom) [reply src/main/java/com/javaproject/restservice/repository/UserRepository.java]: Also why does this need to be an interface?
What is JpaRepository?
and why is it User, String?

**Comment by lynnSlalom** [comment]
You need to add a unit test as well in order to finish the PR out

**Review by lynnSlalom — CHANGES_REQUESTED** [review]

File: src/main/java/com/javaproject/restservice/Application.java (hunk 2 above)
Comment (Jacky-at-Slalom) [reply src/main/java/com/javaproject/restservice/Application.java:6]: it allows me to run the application as a web service in a stand-alone server.

File: pom.xml (hunk 1 above)
Comment (Jacky-at-Slalom) [reply pom.xml]: It breaks my project.  

I think the dependency was added when I was adding a class and did not specify it to be a Java class in intellij.

File: src/main/java/com/javaproject/restservice/entity/User.java (hunk 4 above)
Comment (Jacky-at-Slalom) [reply src/main/java/com/javaproject/restservice/entity/User.java:10]: The @Id annotation marks a field as a primary key field. Im indicating these are my primary keys.

File: src/main/java/com/javaproject/restservice/entity/User.java (hunk 5 above)
Comment (Jacky-at-Slalom) [reply src/main/java/com/javaproject/restservice/entity/User.java:42]: Polymorphism.

Comment (lynnSlalom) [reply src/main/java/com/javaproject/restservice/entity/User.java:42]: Why use it?

File: src/main/java/com/javaproject/restservice/entity/User.java (hunk 4 above)
Comment (lynnSlalom) [reply src/main/java/com/javaproject/restservice/entity/User.java:10]: 👍

File: src/main/java/com/javaproject/restservice/repository/UserRepository.java (hunk 6 above)
Comment (Jacky-at-Slalom) [reply src/main/java/com/javaproject/restservice/repository/UserRepository.java]: The @RepositoryRestResource annotation marks this interface as a REST resource for /users

The interface extends JpaRepository and  The interface also provides all the CRUD implementations and create a Spring bean.

There are  two generic types defined when interface is extended  entity class and its primary key. My entity class is User and the type is String.

**Pull request merged**